app/src/main/java/com/isc/hermes/controller/SearcherController.java

```diff
@@ -0,0 +1,141 @@
+package com.isc.hermes.controller;
+
+import android.os.AsyncTask;
+import android.os.Handler;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.ScrollView;
+import android.widget.SearchView;
+import android.widget.TextView;
+import com.isc.hermes.R;
+import com.isc.hermes.model.searcher.Searcher;
+import com.mapbox.api.geocoding.v5.models.CarmenFeature;
+import java.util.List;
+import java.util.Objects;
+
+
+/**
+ * The SearcherController class is responsible for managing the search functionality
+ * and updating the search results UI.
+ */
+public class SearcherController {
+    private static final long SEARCH_DELAY_MS = 100; // Throttling delay in milliseconds
+    private ScrollView resultsContainer;
+    private LinearLayout searchResultsLayout;
+    private SearchView searchView;
+    private final Searcher searcher;
+    private Handler searchHandler;
+    private Runnable searchRunnable;
+
+    /**
+     * Constructs a new SearcherController.
+     *
+     * @param searcherModel    The Searcher model used for performing the search.
+     * @param resultsContainer The ScrollView that contains the search results UI.
+     * @param searchView       The SearchView widget for inputting search queries.
+     */
+    public SearcherController(Searcher searcherModel, ScrollView resultsContainer, SearchView searchView) {
+        this.searcher = searcherModel;
+        this.resultsContainer = resultsContainer;
+        this.searchResultsLayout = resultsContainer.findViewById(R.id.searchResultsLayout);
+        this.searchView = searchView;
+        this.searchHandler = new Handler();
+        this.searchRunnable = null;
+    }
+
+    /**
+     * Runs the searcher and sets up the search functionality.
+     */
+    public void runSearcher() {
+        resultsContainer.setVisibility(View.INVISIBLE);
+        manageTextFieldSearcherBehaviour();
+    }
+
+    /**
+     * Manages the behavior of the search field.
+     */
+    private void manageTextFieldSearcherBehaviour() {
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                // Add a point on the map to start the navigation
```

Comment:
Delete comment

Reply:
This comment is to remember to add the waypoints on the map later.

Reply:
Good, so leave it as ToDo comment:  //TODO: Add a point on the map to start the navigation